polynote-kernel/src/main/scala/polynote/kernel/ScalaCompiler.scala

```diff
@@ -216,7 +241,13 @@ class ScalaCompiler private (
     lazy val (implicitInputs: List[ValDef], nonImplicitInputs: List[ValDef]) = typedInputs.partition(_.mods.isImplicit)
 
     // a position to encompass the whole synthetic tree
-    private lazy val wrappedPos = Position.transparent(sourceFile, 0, 0, sourceFile.length + 1)
+    private val end = math.max(0, sourceFile.length - 1)
```

Comment:
Here is a switch from transparent positions to opaque positions. I made this change because it turns out that if the tree that defines the class is transparently positioned, then the presentation compiler won't see it at all (I guess that's what they mean by "transparent") and it won't get a `Context`, so none of the other APIs will work at all for any of the code. Making this transparent (and adding some whitespace around the code, which is done in `ScalaInterpreter`) allows the compiler to (in some cases) work properly and give us *some* completion functionality for free.

We still only use this for `Ident` trees, but previously those weren't even working. In the relevant locations I'll explain why we still use our own code (copypasta at that) for completing `Select` trees.